src/inmanta/env.py

```diff
@@ -101,6 +101,36 @@ def _init_env(self) -> bool:
                 return False
             LOGGER.debug("Created a new virtualenv at %s", self.env_path)
 
+        else:
+            # make sure the venv hosts the same python version as the running process
+
+            if sys.platform != "win32":
+                # On UNIX distribs we can check the versions match because the env's version is in the site-packages dir's path
+                binpath = os.path.abspath(os.path.join(self.env_path, "bin"))
+                base = os.path.dirname(binpath)
+                site_packages = os.path.join(
+                    base, "lib", "python%s" % ".".join(str(digit) for digit in sys.version_info[:2]), "site-packages"
+                )
```

Comment:
I think it would make sense to set a `self.site_packages` variable in the constructor. Now we are calculating this value in two different location of the same class.